api/v1beta2/network_types.go

```diff
@@ -94,12 +104,13 @@ var (
 
 // TargetGroupHealthCheck defines health check settings for the target group.
 type TargetGroupHealthCheck struct {
-	Protocol        *string `json:"protocol,omitempty"`
-	Path            *string `json:"path,omitempty"`
-	Port            *string `json:"port,omitempty"`
-	IntervalSeconds *int64  `json:"intervalSeconds,omitempty"`
-	TimeoutSeconds  *int64  `json:"timeoutSeconds,omitempty"`
-	ThresholdCount  *int64  `json:"thresholdCount,omitempty"`
+	Protocol                *string `json:"protocol,omitempty"`
+	Path                    *string `json:"path,omitempty"`
+	Port                    *string `json:"port,omitempty"`
+	IntervalSeconds         *int64  `json:"intervalSeconds,omitempty"`
+	TimeoutSeconds          *int64  `json:"timeoutSeconds,omitempty"`
+	ThresholdCount          *int64  `json:"thresholdCount,omitempty"`
+	UnhealthyThresholdCount *int64  `json:"unhealthyThresholdCount,omitempty"`
```

Comment:
wait is this a new API field?

Reply:
@faiq IIUC the missing field when setting the LB is part of the status[1], updated by controller. Do you think we can drop this status field here moving to the other PR #4849 which is exposing those attributes? Forgive me I am new in the project, trying to understand overall flow. :)

[1] 
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4846/files#diff-aec4bb6b6279e42ec359b4c273734702a032460b7e5fc6ffae4766317a7247b4
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4846/files#diff-819e1fc8b69dba75e117c922715dbad8c4ed5bd8ef5608adbdc00f1eee18a546

Reply:
Let’s keep the API objects as they are for right now and focus on parity between ELB v1 and 2

Reply:
@faiq thanks for the feedback. Commit removed. Using the unhealthy threshold to the constant when creating the TG.

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/7a658ca873bcc1b5630c84f8b4d7de84947d6680/pkg/cloud/services/elb/loadbalancer.go#L338